File: src/core/data_formats/parsing/ninja/angle.ts
const ANGLE_TO_RAD = (2 * Math.PI) / 0x10000;
const RAD_TO_ANGLE = 0x10000 / (2 * Math.PI);

export function angle_to_rad(angle: number): number {
    return angle * ANGLE_TO_RAD;
}

export function rad_to_angle(rad: number): number {
    return Math.round(rad * RAD_TO_ANGLE);
}

File: lib/src/commonMain/kotlin/world/phantasmal/lib/fileFormats/ninja/Angle.kt
package world.phantasmal.lib.fileFormats.ninja

import kotlin.math.PI
import kotlin.math.round

private const val ANGLE_TO_RAD = ((2 * PI) / 0x10000).toFloat()
private const val RAD_TO_ANGLE = (0x10000 / (2 * PI)).toFloat()

fun angleToRad(angle: Int): Float {
    return angle * ANGLE_TO_RAD
}

fun radToAngle(rad: Float): Int {
    return round(rad * RAD_TO_ANGLE).toInt()
}
